core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalLogWriter.java

```diff
@@ -62,11 +66,22 @@
   private UfsJournalGarbageCollector mGarbageCollector;
   /** Whether the journal log writer is closed. */
   private boolean mClosed;
+  /**
+   * Journal entries that have been written successfully to the underlying
+   * {@link DataOutputStream}, but have not been flushed.
+   * The semantic and guarantee provided by the flush operation is
+   * UFS-dependent. For HDFS, Alluxio flushes its journal entries using
+   * hflush. After the successful execution of hflush, the journal entries
+   * have reached (not necessarily persisted) on the minimum number of
+   * datanodes. Therefore, we need a data structure to store the journal
+   * entries that have been written, but not flushed.
+   */
+  private Queue<JournalEntry> mEntriesToFlush;
 
   /**
    * A simple wrapper that wraps a output stream to the current log file.
    */
-  private class JournalOutputStream implements Closeable {
+  protected class JournalOutputStream implements Closeable {
```

Comment:
Because unit test requires access to this. I'll remove the `protected` keyword and use package default.

Reply:
Done.

Reply:
It can stay private - we only use it via Whitebox and for `null` checking, and neither requires visibility of the class. In the unit test, we can refer to the `JournalOutputStream` using type `Object`.

Reply:
Done.